packages/repository-json-schema/src/__tests__/integration/build-schema.integration.ts

```diff
@@ -1001,5 +1001,79 @@ describe('build-schema', () => {
         expect(excludeNothingSchema.title).to.equal('Product');
       });
     });
+
+    context('optional properties when option "optional" is set', () => {
+      @model()
+      class Product extends Entity {
+        @property({id: true, required: true})
+        id: number;
+
+        @property({required: true})
+        name: string;
+
+        @property()
+        description: string;
+      }
+
+      it('makes one property optional when the option "optional" includes one property', () => {
+        const originalSchema = getJsonSchema(Product);
+        expect(originalSchema.required).to.deepEqual(['id', 'name']);
+        expect(originalSchema.title).to.equal('Product');
+
+        const optionalIdSchema = getJsonSchema(Product, {optional: ['id']});
+        expect(optionalIdSchema.required).to.deepEqual(['name']);
+        expect(optionalIdSchema.title).to.equal('ProductOptional[id]');
+      });
+
+      it('makes multiple properties optional when the option "optional" includes multiple properties', () => {
+        const originalSchema = getJsonSchema(Product);
+        expect(originalSchema.required).to.deepEqual(['id', 'name']);
+        expect(originalSchema.title).to.equal('Product');
+
+        const optionalIdAndNameSchema = getJsonSchema(Product, {
+          optional: ['id', 'name'],
+        });
+        expect(optionalIdAndNameSchema.required).to.equal(undefined);
+        expect(optionalIdAndNameSchema.title).to.equal(
+          'ProductOptional[id,name]',
+        );
+      });
+
+      it('doesn\'t make properties optional when the option "optional" includes no properties', () => {
```

Comment:
Nitpick: we can use ` to quote the string to avoid escaping, such as:
```
`doesn't make properties optional when the option "optional" includes no properties`
```